astropy/table/jsviewer.py

```diff
@@ -84,12 +84,19 @@ class Conf(_config.ConfigNamespace):
 
 
 HTML_JS_SCRIPT = """
+jQuery.extend( jQuery.fn.dataTableExt.oSort, {{
+    "optionalnum-pre": function (a) {{ var b=parseFloat(a); if (isNaN(b)) return Number.NEGATIVE_INFINITY; else return b;}},
+    "optionalnum-asc": function (a,b) {{ return ((a < b) ? -1 : ((a > b) ? 1 : 0)); }},
+    "optionalnum-desc": function (a,b) {{ return ((a < b) ? 1 : ((a > b) ? -1 : 0)); }}
+}});
+
 $(document).ready(function() {{
     $('#{tid}').dataTable({{
      "iDisplayLength": {display_length},
      "aLengthMenu": {display_length_menu},
      "bJQueryUI": true,
-     "sPaginationType": "full_numbers"
+     "sPaginationType": "full_numbers",
+     "aoColumnDefs": [{{"sType": "optionalnum", "aTargets": {sort_columns}}}]
```

Comment:
agreed, it is dead... @embray on looking again, I think I tried, it had too many bugs, then I gave up.